src/metabase/query_processor/middleware/parameters/mbql.clj

```diff
@@ -94,12 +94,15 @@
 
 (defn- update-breakout-unit
   [query
-   {[_dimension [_field target-field-id {:keys [base-type temporal-unit]}]] :target
+   {[_dimension [_field target-field-id {:keys [base-type temporal-unit]}] dim-opts] :target
     :keys [value] :as _param}]
   (let [new-unit (keyword value)
         base-type (or base-type
                       (when (integer? target-field-id)
-                        (:base-type (lib.metadata/field (qp.store/metadata-provider) target-field-id))))]
+                        (:base-type (lib.metadata/field (qp.store/metadata-provider) target-field-id))))
+        stage-path (if-let [stage-number (:stage-number dim-opts)]
+                     (mbql.u/stage-path (:query query) stage-number)
+                     [])]
```

Comment:
Perhaps `mbql.u/stage-path` itself should respond to a stage number of `nil` with `[]`, ie. the last stage/outermost query?

Reply:
That's a good idea, thanks!